Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -613,6 +613,115 @@ TEST_CASE_METHOD(ReplicatorAPITest, "Is Document Pending", "[C][Push]") {
 }
 #endif
 
+#ifdef COUCHBASE_ENTERPRISE
+
+TEST_CASE_METHOD(ReplicatorAPITest, "Pending Document IDs Non-Existent Collection", "[C][Push]") {
+    ExpectingExceptions x;
+    // Create collection in the db and import documents
+    C4CollectionSpec Kyber = {"kyber"_sl, "crystal"_sl};
+    auto collKyber = createCollection(db, Kyber);
+    importJSONLines(sFixturesDir + "names_100.json", collKyber);
+
+    // Create the collection in the database to be replicated to
+    createDB2();
+    createCollection(db2, Kyber);
+
+    // Create collection spec but do not create collection in db
+    C4CollectionSpec Republic = {"republic"_sl, "galactic"_sl};
+
+    // Set options for the replication
+    C4Error err;
+    repl::C4ReplParamsCollection params{Kyber};
+    params.push = kC4OneShot;
+    params.pull = kC4Disabled;
+    params.callbackContext = this;
+    params.socketFactory = _socketFactory;
+
+    _repl = c4repl_newLocal(db, (C4Database*)db2, params, ERROR_INFO(err));
+    REQUIRE(_repl);
+
+    C4SliceResult encodedDocIDs = c4repl_getPendingDocIDs(_repl, Republic, &err);
+    CHECK(err.code == kC4ErrorNotOpen);
+    c4slice_free(encodedDocIDs);
+
+    c4repl_start(_repl, false);
+    REQUIRE_BEFORE(5s, c4repl_getStatus(_repl).level == kC4Stopped);
+    encodedDocIDs = c4repl_getPendingDocIDs(_repl, Republic, &err);
+    CHECK(err.code == kC4ErrorNotOpen);
+}
+
+TEST_CASE_METHOD(ReplicatorAPITest, "Pending Document IDs Multiple Collections", "[C][Push]") {
+    // Create collections and import documents to them
+    C4CollectionSpec Council = { "council"_sl, "jedi"_sl };
+    C4CollectionSpec Federation = { "federation"_sl, "trade"_sl };
+    auto collCouncil = createCollection(db, Council);
+    auto collFederation = createCollection(db, Federation);
+    importJSONLines(sFixturesDir + "names_100.json", collCouncil);
+    importJSONLines(sFixturesDir + "wikipedia_100.json", collFederation);
+
+    // Create the collections in the database to be replicated to
+    createDB2();
+    createCollection(db2, Council);
+    createCollection(db2, Federation);
+
+    // Set options for the replication
+    // Replicating only the Council collection
+    FLSliceResult options {};
```

Comment:
Where did you use options?

Reply:
Ah sorry, that was left over from when the test had multiple sections